rust/lance/src/index/vector/ivf.rs

```diff
@@ -631,8 +631,21 @@ pub async fn build_ivf_pq_index(
         lance_index::vector::pq::num_centroids(pq_params.num_bits as u32),
     ) * ivf_params.sample_rate;
 
-    // TODO: only sample data if training is necessary.
-    let mut training_data = maybe_sample_training_data(dataset, column, sample_size_hint).await?;
+    let mut training_data = if ivf_params.centroids.is_none() {
+        let start = std::time::Instant::now();
+        log::info!(
+            "Loading training data for IVF. Sample size: {}",
+            sample_size_hint
+        );
+        let data = Some(maybe_sample_training_data(dataset, column, sample_size_hint).await?);
+        log::info!(
+            "Finished loading training data in {:02} seconds",
+            start.elapsed().as_secs_f32()
+        );
+        data
```

Comment:
nit: maybe just add the timer in `maybe_sample_training_data`?

Reply:
With the narrative logs, I like that they are all in 1 function. 🤷